catalogs/catalog-jdbc-mysql/src/main/java/com/datastrato/gravitino/catalog/mysql/operation/MysqlTableOperations.java

```diff
@@ -102,13 +109,71 @@ public JdbcTable load(String databaseName, String tableName) throws NoSuchTableE
                       }
                     }
                   }))
+          .withIndexes(indexes.toArray(new Index[0]))
           .withAuditInfo(AuditInfo.EMPTY)
           .build();
     } catch (SQLException e) {
       throw exceptionMapper.toGravitinoException(e);
     }
   }
 
+  private List<Index> getIndexes(String databaseName, String tableName, DatabaseMetaData metaData)
+      throws SQLException {
+    List<Index> indexes = new ArrayList<>();
+    // 1.Get primary key information
+    Map<String, Set<String>> primaryKeyGroupByName = new HashMap<>();
```

Comment:
I think probably you can use `Multiplemap` to simplify the code here https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Multimap.html.